iidm/iidm-tck/src/test/java/com/powsybl/iidm/network/tck/AbstractAreaTest.java

```diff
@@ -0,0 +1,487 @@
+        assertEquals(-602.6, controlAreaA.getAcInterchangeTarget().orElseThrow());
+        assertEquals(Set.of(vlgen, vlhv1), controlAreaA.getVoltageLevels());
+        assertEquals(2, controlAreaA.getAreaBoundaryStream().count());
+
+        assertEquals("ControlArea_B", controlAreaB.getId());
+        assertEquals("Control Area B", controlAreaB.getOptionalName().orElseThrow());
+        assertEquals(CONTROL_AREA_TYPE, controlAreaB.getAreaType());
+        assertEquals(+602.6, controlAreaB.getAcInterchangeTarget().orElseThrow());
+        assertEquals(Set.of(vlhv2, vlload), controlAreaB.getVoltageLevels());
+        assertEquals(2, controlAreaB.getAreaBoundaryStream().count());
+
+        assertEquals("Region_AB", regionAB.getId());
+        assertEquals("Region AB", regionAB.getOptionalName().orElseThrow());
+        assertEquals(REGION_AREA_TYPE, regionAB.getAreaType());
+        assertFalse(regionAB.getAcInterchangeTarget().isPresent());
+        assertEquals(Set.of(vlgen, vlhv1, vlhv2, vlload), regionAB.getVoltageLevels());
+        assertEquals(0, regionAB.getAreaBoundaryStream().count());
+    }
+
+    @Test
+    public void testSetterGetter() {
+        controlAreaA.setAcInterchangeTarget(123.0);
+        assertTrue(controlAreaA.getAcInterchangeTarget().isPresent());
+        assertEquals(123.0, controlAreaA.getAcInterchangeTarget().getAsDouble());
+        controlAreaA.setAcInterchangeTarget(Double.NaN);
+        assertTrue(controlAreaA.getAcInterchangeTarget().isEmpty());
+    }
+
+    @Test
+    public void testChangesNotification() {
+        // Changes listener
+        NetworkListener mockedListener = Mockito.mock(DefaultNetworkListener.class);
+        // Add observer changes to current network
+        network.addListener(mockedListener);
+
+        // Get initial values
+        double oldValue = controlAreaA.getAcInterchangeTarget().orElseThrow();
+        // Change values
+        controlAreaA.setAcInterchangeTarget(Double.NaN);
+
+        // Check update notification
+        Mockito.verify(mockedListener, Mockito.times(1))
+                .onUpdate(controlAreaA, "acInterchangeTarget", VariantManagerConstants.INITIAL_VARIANT_ID, oldValue, Double.NaN);
+
+        // Change values
+        controlAreaA.setAcInterchangeTarget(123.4);
+
+        // Check update notification
+        Mockito.verify(mockedListener, Mockito.times(1))
+                .onUpdate(controlAreaA, "acInterchangeTarget", VariantManagerConstants.INITIAL_VARIANT_ID, Double.NaN, 123.4);
+
+        // At this point
+        // no more changes is taking into account
+
+        // Simulate exception for onUpdate calls
+        Mockito.doThrow(new PowsyblException()).when(mockedListener)
+                .onUpdate(controlAreaA, "acInterchangeTarget", VariantManagerConstants.INITIAL_VARIANT_ID, oldValue, 123.4);
+
+        // Case when same value is set
+        controlAreaA.setAcInterchangeTarget(123.4);
```

Comment:
Isn't an assertion missing here?

Reply:
the relevant assertion is the one at the very end: mockito's "no interaction wanted".

there was a bug in the test, fixed now, see below comment HERE--->

```java
        // Case when same value is set
        controlAreaA.setAcInterchangeTarget(123.4);
        // Case when no listener is registered
        network.removeListener(mockedListener);
        controlAreaA.setAcInterchangeTarget(456.7); // HERE---> was 123.4 before, so not really testing listener removal, this is fixed now

        // Check no notification
        Mockito.verifyNoMoreInteractions(mockedListener);

```

Reply:
OK, that's clear!